### PR TITLE
More test fixes

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: test/kevent.c [hunk 1]
```diff
@@ -43,8 +43,10 @@ kevent_get(struct kevent *kev, int kqfd)
     struct kevent buf;
     int nfds;
 
-    if (kev == NULL)
-       kev = &buf;
+    if (kev == NULL) {
+        memset(&buf, 0, sizeof(buf));
```
Comment (arr2036) [inline test/kevent.c:47]: Why is this needed? The majority of applications won't initialise the kevent buffers before passing them in.  If there's uninitialised fields then kevent should be fixed to initialise them

File: test/common.h [hunk 2]
```diff
@@ -22,7 +22,10 @@
 # include <err.h>
 #else
 # define err(rc,msg,...) do { perror(msg); exit(rc); } while (0)
-# define errx(rc,msg,...) do { puts(msg); exit(rc); } while (0)
+# define errx(rc, msg, ...) do { \
+    fprintf(stderr, msg"\n", ##__VA_ARGS__);\
```
Comment (arr2036) [inline test/common.h:26]: Looks fine, but we really should have a similar fix for err too.

File: test/user.c [hunk 3]
```diff
@@ -62,7 +62,7 @@ test_kevent_user_get_hires(struct test_context *ctx)
     kev.fflags &= ~NOTE_FFCTRLMASK;
     kev.fflags &= ~NOTE_TRIGGER;
     kev.flags = EV_CLEAR;
-    kevent_get_hires(&ret, ctx->kqfd);
+    kevent_get_hires(&ret, ctx->kqfd, 5);
```
Comment (arr2036) [inline test/user.c:65]: This increases the timeout massively, from 0.0005 seconds to 5 seconds?  Why not just pass a timespec in?

File: test/kevent.c [hunk 4]
```diff
@@ -53,6 +53,31 @@ kevent_get(struct kevent *kev, int kqfd)
...
 }
 
+/**
+ * Retrieve a single kevent, specifying a maximum time to wait for it.
+ * Result:
+ * 0 = Timeout
+ * 1 = Success
+ */
+int
+kevent_get_timeout(struct kevent *kev, int fd, int seconds)
```
Comment (arr2036) [inline test/kevent.c:63]: Should be using a timespec.  Seconds is way too long for CIT

File: test/common.h (hunk 2 above)
Comment (jpereira) [reply test/common.h:26]: yeah! good idea.
